git-arc

```diff
@@ -203,33 +231,62 @@ restore_head()
 
 gitarc::create()
 {
-    local commit commits orig prev reviewers
+    local chash commit _commits commits doprompt list o orig prev reviewers
+    local subscribers
 
-    while getopts r: o; do
+    list=
+    if eval $(git config --bool --default false --get arc.list); then
+        list=1
+    fi
+    doprompt=1
+    while getopts lr:s: o; do
         case "$o" in
+        l)
+            list=1
+            ;;
         r)
             reviewers="$OPTARG"
             ;;
+        s)
+            subscribers="$OPTARG"
+            ;;
         *)
             err_usage
             ;;
         esac
-        shift $((OPTIND-1))
     done
+    shift $((OPTIND-1))
 
-    orig=$(save_head)
-    prev=""
+    commits=""
     for chash in $@; do
         # Convert each commit or commit range into a list of hashes.
-        commits=$(git rev-parse "${chash}")
+        _commits=$(git rev-parse "${chash}")
         if ! git cat-file -e "${chash}"'^{commit}' >/dev/null 2>&1; then
-            commits=$(git rev-list $commits | tail -r)
-            [ -n "$commits" ] || err "invalid commit ID ${chash}"
+            _commits=$(git rev-list $commits | tail -r)
+            [ -n "$_commits" ] || err "invalid commit ID ${chash}"
         fi
+        commits="${commits} ${_commits}"
+    done
+
+    if [ "$list" ]; then
         for commit in ${commits}; do
-            create_one_review "$commit" "$reviewers" "$prev"
-            prev=$(commit2diff "$commit")
+            git --no-pager show --oneline --no-patch $commit
         done
+        if ! prompt; then
+            return
+        fi
+        doprompt=
+    fi
+
+    orig=$(save_head)
+    prev=""
+    for commit in ${commits}; do
+        if create_one_review "$commit" "$reviewers" "$subscribers" "$prev" \
+                             "$doprompt"; then
+            prev=$(commit2diff "$commit")
+        else
+            prev=""
```

Comment:
Removing this else would instead stitch together the commits that aren't rejected by the user into a single stack.  I'm not sure what semantics we want (should the user rejecting a diff just terminate entirely, or just skip that revision, and if so, should skipping create a "hole" and a new stack like this patch does, or still create a single stack)